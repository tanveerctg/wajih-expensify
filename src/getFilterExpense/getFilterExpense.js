  import moment from 'moment';

  export default (exp,{text='',startDate,endDate,sortBy})=>{

    let hold=exp.filter(expense=>{
    
      const txtMatch =expense.description.toLowerCase().includes(text.toLowerCase());
      const startingDate = startDate ? expense.createdat>= moment(startDate).valueOf() : true;
      const endingDate = endDate ? expense.createdat <= moment(endDate).valueOf() : true;
   
    if(txtMatch && startingDate && endingDate){
      return expense;
    }
    });
  
    if(sortBy==='amount'){
  
      hold=hold.sort((a,b)=>(a.amount<b.amount?1:-1))
    }else if(sortBy==='date'){
      
      hold=hold.sort((a,b)=>(a.createdat<b.createdat?1:-1))
    }
    let sum=0;
    const totalExpense=hold.forEach(item=>{
      sum=parseInt(item.amount)+sum;
    })

    return { hold:hold,
             totalExpense:sum
           };
  
   
  }
