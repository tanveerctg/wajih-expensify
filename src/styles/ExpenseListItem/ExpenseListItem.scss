
//ExpenseListItem
.expenseListItem{
  display:grid;
  grid-template-columns: 1fr auto;
  padding:15px;
  align-items:center;
  color:black;
}
.link__style{
  text-decoration: none;
}
.expenseListItem__1st{
  display:grid;
  grid-gap:8px;
}
.expenseListItem__2nd{
  display:grid;
  align-items:center;
}
.expenseListItem:hover{
  background-color: #f4f1f5;
}