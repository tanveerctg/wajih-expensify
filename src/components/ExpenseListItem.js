import React from 'react';
import { Link } from 'react-router-dom';
import RemoveItem from './RemoveItem';
import { connect } from 'react-redux';
import getFilterExpense from '../getFilterExpense/getFilterExpense';

const ExpenseListItem=({id,description,createdat,amount})=>(
  <div>
    <Link to={`edit/${id}`} ><h3>{description}</h3></Link>
    <p>created at-{createdat}</p>
    <p>amount -{amount}</p>
    <RemoveItem id={id}/>
    <br/>
  </div>
);

export default ExpenseListItem;

