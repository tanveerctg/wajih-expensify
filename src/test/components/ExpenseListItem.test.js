import toJson from 'enzyme-to-json';
import React from 'react';
import { shallow } from 'enzyme';
import ExpenseListItem from  '../../components/ExpenseListItem';
import setupTests from '../setupTests';
import getFilterExpense from '../getFilterExpense/getFilterExpense.test.js';

test("should render ExpenseListItem correctly",()=>{
  const wrapper=shallow(<ExpenseListItem { ...getFilterExpense[0] }/>);
  expect(toJson(wrapper)).toMatchSnapshot();
})