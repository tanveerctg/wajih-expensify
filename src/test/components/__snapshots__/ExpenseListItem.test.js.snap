// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListItem correctly 1`] = `
<div>
  <Link
    replace={false}
    to="edit/1"
  >
    <h3>
      January
    </h3>
  </Link>
  <p>
    created at-
    1300
  </p>
  <p>
    amount -
    2000
  </p>
  <Connect(RemoveItem)
    id="1"
  />
  <br />
</div>
`;
